node/src/components/fetcher.rs

```diff
@@ -524,15 +531,23 @@ where
                 }
                 None => self.failed_to_get_from_storage(effect_builder, id, peer),
             },
-            Event::GotRemotely { item, source } => {
+            Event::GotRemotely {
+                merkle_tree_hash_activation: _,
```

Comment:
Unfortunately, it is needed for the sake of `impl Display`, like here: https://github.com/casper-network/casper-node/pull/2531/files#diff-eef5ec27d08d2fabcc4844fb7bb058c0e88ccb2d2f9e352facd842b7b9666554R134-R149

That was the simplest approach I came up with to be able to present the correct hash in the `Display`.

I will happily accept any other approach that doesn't require substantial refactoring. Alternatively, we may spawn a ticket to handle that in the future.

Reply:
Hmm… maybe adding an `id` field instead of `merkle_tree_hash_activation` would work? We'd just compute the hash before generating the event: Then we wouldn't need a field that's specific to one event type.
Anyway, I'm not sure about it. It's not a blocker.

Reply:
Having the `id` in the event could work, but also has drawbacks:
1. event creators need to have access to `merkle_tree_hash_activation` - even if they don't care (we can solve it by having an `Option<T::Id>`
2. even with the Option, we'll end-up having duplicated information in the Event, so the `handle_event` can access both `item.id()` or `T::Id`. We have two items that should match, but we can't enforce that in the code.

I'll leave it like it is with a `TODO` comment. I have a feeling that sooner or later we'd need to refactor out this `merkle_tree_hash_activation` tramp data.

Reply:
Comment added in https://github.com/casper-network/casper-node/pull/2531/commits/efabc750a6da1dc3b8d15d7e421b1bf2c5fee74a